### PR TITLE
Add RPC failover options

### PR DESCRIPTION
**Pull request by Sliman4**
I thought about creating a new page for all failover options in all environments, but `near-jsonrpc-client-rs` doesn't have this feature, `wallet-selector` only has a feature request issue, couldn't find it in other packages, either, so I guess the providers page is a good place for this for now.

Also fixed a link on that page that didn't scroll to the highlighted content.

### CODE REVIEW

**Comment by frol** [comment]
@Sliman4 Sorry for jumping in, but I'd like to invite you to participate in the [Race of Sloths](https://race-of-sloths.com) - a fun and gamified open-source contributions program. Consider mentioning @race-of-sloths user in your github comment or PR description to join!

See how the flow works here: https://github.com/near/nearcore/pull/11778

**Comment by Sliman4** [comment]
@race-of-sloths

**Comment by race-of-sloths** [comment]
@Sliman4 Thank you for your contribution! Your pull request is now a part of the [Race of Sloths](https://race-of-sloths.com)!
New Sloth joined the Race! Welcome!

[<picture>
    <source media="(prefers-color-scheme: dark)" srcset="https://badge.race-of-sloths.com/Sliman4?type=bot&theme=dark">
    <source media="(prefers-color-scheme: light)" srcset="https://badge.race-of-sloths.com/Sliman4?type=bot&theme=light">
    Shows profile picture for the author of the PR
</picture>](https://race-of-sloths.com/profile/Sliman4)


<details><summary>Current status: <i>waiting for scoring</i></summary>

We're waiting for maintainer to score this pull request with `@race-of-sloths score [0,1,2,3,5,8,13]` command
</details><details><summary>What is the Race of Sloths</summary>
<p>
Race of Sloths is a friendly competition where you can participate in challenges and compete with other open-source contributors within your normal workflow

For contributors:

- Tag @race-of-sloths inside your pull requests
- Wait for the maintainer to review and score your pull request
- Check out your position in the [Leaderboard](https://race-of-sloths.com/leaderboard)
- Keep weekly and monthly streaks to reach higher positions
- Boast your contributions with a dynamic picture of your [Profile](https://race-of-sloths.com/profile/Sliman4)

For maintainers:
- Score pull requests that participate in the Race of Sloths
- Engage contributors with fair scoring and fast responses so they keep their streaks
- Promote the Race to the point where the Race starts promoting you
- Grow the community of your contributors

Feel free to check [our website](https://race-of-sloths.com) for additional details!
</p>
</details> 

<details>
<summary>Bot commands</summary>

- For contributors
  - **Include a PR:** `@race-of-sloths include` to enter the Race with your PR
- For maintainers:
  - **Assign points:** `@race-of-sloths score [1/2/3/5/8/13]` to award points based on your assessment.
  - **Reject this PR:** `@race-of-sloths exclude` to send this PR back to the drawing board.
  - **Exclude repo:** `@race-of-sloths pause` to stop bot activity in this repo until `@race-of-sloths unpause` command is called

</details>

File: docs/2.build/4.web3-apps/integrate-contracts.md
Comment (bucanero) [inline docs/2.build/4.web3-apps/integrate-contracts.md:60]: this `id` doesn't look to be used anywhere.

File: docs/5.api/rpc/providers.md
Comment (bucanero) [inline docs/5.api/rpc/providers.md:49]:
```suggestion
In `near-api-js` you can use [`FailoverRpcProvider`](../../4.tools/near-api-js/quick-reference.md#rpc-failover) to automatically switch RPC providers when one provider is experiencing downtime, or implement an RPC selection widget that allows users to add their own RPC provider.
```

File: docs/5.api/rpc/providers.md
Comment (bucanero) [inline docs/5.api/rpc/providers.md:51]:
```suggestion
As a user, if a dApp or wallet doesn't support RPC failover and the primary provider is down, you can use an RPC Selector browser extension to redirect all requests to an RPC provider of your choice.
```

**Review by bucanero — CHANGES_REQUESTED** [review]
Please check my comments. 👍

File: docs/2.build/4.web3-apps/integrate-contracts.md
Comment (Sliman4) [reply docs/2.build/4.web3-apps/integrate-contracts.md:60]: It's used in "If you want to use a custom RPC provider with NEAR Wallet Selector, [check this example](https://docs.near.org/build/web3-apps/integrate-contracts#setting-customs-rpc-endpoints)." in https://docs.near.org/api/rpc/providers (not changed in this pr), but when you click the link, it doesn't scroll to the example because the id didn't exist

File: docs/5.api/rpc/providers.md
Comment (Sliman4) [reply docs/5.api/rpc/providers.md:51]: Do you think it's worth mentioning the extension if there's no link? This extension with 3 users is surprisingly well googleable using "rpc selector" term, but I'm not sure if anyone is going to even try to find it. Personally, I never try googling anything with such a generic name. The whole reason I made this PR is that people in the dev community found it useful ([1](https://t.me/neardev/51431), [2](https://t.me/neardev/51432)), and removing the link defeats the entire purpose. If the issue is that it links to a third-party-owned repo, I can transfer ownership, I'm not actively maintaining it anyway

Comment (bucanero) [reply docs/5.api/rpc/providers.md:51]: We can't reference such a tool without a proper security audit, as this could impact user's transactions. (Even after a repository transfer, an audit would be required)

**Review by bucanero — APPROVED** [review]

**Comment by race-of-sloths** [comment]
### 🔄 The PR has been merged.
>[!IMPORTANT]
This pull request is a part of the Race of Sloths and has not been scored yet. Scoring will close in 24 hours! 🕰️

**Comment by race-of-sloths** [comment]
### ✅ PR is finalized! 
Your contribution is much appreciated with a final score of 2!
You have received 30 (20 base + 10 weekly bonus) Sloth points for this contribution


Another weekly streak completed, well done @Sliman4! To keep your weekly streak and get another bonus make pull request next week! Looking forward to see you in race-of-sloths

**Pull request merged**
